packages/gestalt/src/HelpButton.js

```diff
@@ -0,0 +1,239 @@
+
+  const textElement =
+    typeof text === 'string' ? (
+      <Text align="start">{text}</Text>
+    ) : (
+      <span className={textColorOverrideStyles}>{text}</span>
+    );
+
+  const PopoverElement = (
+    <Popover
+      id={popoverId}
+      accessibilityLabel={accessibilityPopoverLabel}
+      anchor={tapAreaRef.current}
+      onDismiss={toggleView}
+      onKeyDown={handleKeyDownPopover}
+      idealDirection={idealDirection ?? 'down'}
+      positionRelativeToAnchor={!isWithinScrollContainer}
+    >
+      <Box padding={5} rounding={4} height="auto">
+        <Box
+          ref={firstElementRef}
+          tabIndex={0}
+          onBlur={() => {
+            if (!link?.href) {
+              setFocused(false);
+            }
+          }}
+        >
+          {textElement}
+        </Box>
+        {link?.href && (
+          <Box width="100%" display="block" marginTop={3}>
+            <Text>
+              <Link
+                accessibilityLabel={link?.accessibilityLabel}
+                externalLinkIcon={link?.externalLinkIcon}
+                href={link?.href}
+                onClick={link?.onClick}
+                onFocus={() => setFocused(true)}
+                onBlur={() => setFocused(false)}
+                ref={link?.ref}
+                target={link?.target ?? 'blank'}
+                underline="always"
+              >
+                {link?.text}
+              </Link>
+            </Text>
+          </Box>
+        )}
+      </Box>
+    </Popover>
+  );
+
+  return (
+    // The only purpose of this Flex is to make zIndex work (Tooltip over Popover).
+    <Flex alignItems="center" justifyContent="center" flex="none">
+      <Tooltip
+        accessibilityLabel=""
+        text={tooltipMessage}
+        idealDirection={idealDirection}
+        zIndex={tooltipZIndex}
+      >
+        <TapArea
+          accessibilityExpanded={open}
+          accessibilityControls={popoverId}
+          accessibilityLabel={accessibilityLabel}
+          fullWidth={false}
+          onTap={onHandleTap}
+          ref={tapAreaRef}
+          role="button"
+          rounding="circle"
+          onBlur={() => setFocused(false)}
+          onFocus={() => setFocused(true)}
+          onMouseEnter={() => setHovered(true)}
+          onMouseLeave={() => setHovered(false)}
+          onKeyDown={handleKeyDownTapArea}
+        >
+          <Box
+            width={16}
+            height={16}
+            rounding="circle"
+            display="flex"
+            alignItems="center"
+            justifyContent="center"
+            color={bgIconColor}
+          >
+            <Icon accessibilityLabel="" color="inverse" icon="question-mark" size={8} />
+          </Box>
+        </TapArea>
+      </Tooltip>
+      {open &&
+        (isWithinScrollContainer ? (
+          // This Layer is handling the Popover positioning on Scrollable containers
+          <Layer zIndex={zIndexWrapper}>{PopoverElement}</Layer>
```

Comment:
Not sure about this

          <Layer zIndex={zIndexWrapper}>{PopoverElement}</Layer>

or

          <Layer zIndex={zIndexWrapper}><PopoverElement /></Layer>

Reply:
I usually do popoverElement, not PopoverElement. I think uppercase allows <PopoverElement />, lowercase is {popoverElement} 


but not sure. https://www.developerway.com/posts/react-re-renders-guide#part3.1